### ISSUE TITLE
Custom Barcode Arrangements: kit is not optional

### ISSUE THREAD

**Issue by danpal96** [nanoporetech/dorado#1015]
# Issue Report

## Please describe the issue:

According to:
https://github.com/nanoporetech/dorado/blob/release-v0.7/documentation/CustomBarcodes.md
The kit option in the barcode arrangement file is optional but it throws an error if it does not have it

## Steps to reproduce the issue:
```bash
dorado demux test_D0.bam --barcode-arrangement test_barcode_arrangement.txt --barcode-sequences test_barcodes.fasta --verbose --output-dir test
```

## Run environment:
- Dorado version: 0.7.3+6e6c45c
- Operating system: Ubuntu 20.04.6 LTS x86_64
- Dataset to reproduce, if applicable (small subset of data to share as a pod5 to reproduce the issue): [test.zip](https://github.com/user-attachments/files/16880661/test.zip)

## Logs
```
[2024-09-04 16:36:16.078] [info] Running: "demux" "basecalling/test_D0.bam" "--barcode-arrangement" "../../resources/dorado/test_barcode_arrangement.txt" "--barcode-sequences" "../../resources/dorado/test_barcodes.fasta" "--verbose" "--output-dir" "test"
[2024-09-04 16:36:16.078] [info] num input files: 1
[2024-09-04 16:36:16.078] [debug] > barcoding threads 15, writer threads 1
[2024-09-04 16:36:16.106] [info] > starting barcode demuxing
terminate called after throwing an instance of 'std::out_of_range'
  what():  [error] key "kit" not found
 --> ../../resources/dorado/test_barcode_arrangement.txt
   |
 1 | [arrangement]
   | ~~~~~~~~~~~~~ in this table
fish: “dorado demux basecalling/test_D…” terminated by signal SIGABRT (Abort)
```

**Comment by CGD-Helix** [comment]
Hi @danpal96,

I believe your arrangement file should be a `.toml` file and not a `.txt` file as you have it. That may fix your issue?

**Comment by danpal96** [comment]
Hi @CGD-Helix 

Here is an arrangement file that works: [test_barcode_arrangement_fixed.txt](https://github.com/user-attachments/files/16910386/test_barcode_arrangement_fixed.txt)
As you can see the only difference is that I added the line `kit = "test"`

Is this a bug or the documentation saying that it is optional is wrong?

**Comment by CGD-Helix** [comment]
Hi @danpal96,

Ah yes, it seems to only work if something is specified for the kit key, even if it is an empty string. Possibly an error in the documentation from ONT on this being an optional parameter!

**Comment by malton-ont** [comment]
Hi @danpal96, @CGD-Helix,

Thanks for reporting this. This looks like a documentation error - I believe this should be more like `Required - may be an empty string`. We'll look into getting that updated.

**Comment by malton-ont** [comment]
This has been addressed in dorado 0.8.0 - the `[kit]` entry is now optional.
